src/modal/modal.js

```diff
@@ -266,6 +290,8 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.transition'])
         openedWindows.top().value.modalDomEl = modalDomEl;
         body.append(modalDomEl);
         body.addClass(OPENED_MODAL_CLASS);
+
+        if (scrollBarWidth) { setScrollBar(); }
```

Comment:
You should check if backdrop index is zero here. If you open modal, while another modal is open, it will shift body and fixed elements for another scrollbar width.